include/functional/detail/meta.hpp

```diff
@@ -0,0 +1,161 @@
+                                                          : TU_name.data() + (TU_name.size() - TU_name_bound - 1),
+                          std::min(TU_name.size() - 1, TU_name_bound)};
+
+    s += prefix.size();
+    while (true) {
+      std::size_t i = sv.find(anon, s);
+      if (i != std::string_view::npos) {
+        append(result, sv.substr(s, i - s));
+        append(result, std::string_view("(anonymous namespace in "));
+        append(result, file);
+        append(result, std::string_view(")"));
+      } else {
+        append(result, sv.substr(s, sv.size() - s - 2));
+        break;
+      }
+      s = i + anon.size();
+    };
+
+    return result;
+  }
+
+  static constexpr auto _slice = apply();
+  static constexpr std::string_view value{_slice.data.data(), _slice.end};
+};
+
+template <typename T> [[nodiscard]] static constexpr auto _make_sortkey()
+{
+  return _normalized_name<std::to_array(__BASE_FILE__), std::to_array(__PRETTY_FUNCTION__)>::value;
+}
+template <typename T> constexpr inline std::string_view type_sortkey_v = _make_sortkey<T>();
+
+// NOTE Normalized order of types - order based on type_sortkey_v
+template <typename... Ts> struct normalized final {
+  static constexpr std::size_t N = sizeof...(Ts);
+
+  struct _uniqued final {
+    std::array<std::size_t, N> indices;
+    std::size_t size;
+  };
+
+  [[nodiscard]] static constexpr auto _indices() noexcept
+  {
+    std::array<std::string_view, sizeof...(Ts)> names{type_sortkey_v<Ts>...};
+    std::array<std::size_t, sizeof...(Ts)> indices;
+    std::generate(indices.begin(), indices.end(), [n = 0]() mutable -> std::size_t { return n++; });
+    auto const less = [v = &names](std::size_t i, std::size_t j) constexpr { return (*v)[i] < (*v)[j]; };
+    std::sort(indices.begin(), indices.end(), less);
+    auto const equal = [v = &names](std::size_t i, std::size_t j) constexpr { return (*v)[i] == (*v)[j]; };
+    auto const end = std::unique(indices.begin(), indices.end(), equal);
+    return _uniqued{indices, static_cast<std::size_t>(end - indices.begin())};
+  }
+
+  static constexpr _uniqued _indices_v = _indices();
+
+  template <template <typename...> typename F, std::size_t... Is>
+  static constexpr auto _normalized_f(std::index_sequence<Is...> const &)
+      -> F<select_nth_t<_indices_v.indices[Is], Ts...>...>;
+
+  // How many unique types
+  static constexpr std::size_t size = _indices_v.size;
+
+  // Apply a given template on a normalized list of types
+  template <template <typename...> typename F>
+  using apply = decltype(_normalized_f<F>(std::make_index_sequence<size>()));
+};
+
+template <typename... T> static constexpr bool is_superset_of = false;
+template <template <typename...> typename F, typename... Ts, typename... Tu>
+static constexpr bool is_superset_of<F<Ts...>, F<Tu...>> = (... && type_one_of<Tu, Ts...>);
+
+template <typename... Ts> struct is_normal final {
+  template <typename... Tx> struct _ts final {};
```

Comment:
why is _ts an inner struct, you don't need it to be namespaced in is_normal's template, you're genning too many types :)